consensus/metrics.go

```diff
@@ -0,0 +1,64 @@
+package consensus
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/discard"
+	prometheus "github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+type Metrics struct {
+	Validators metrics.Gauge
+
+	Rounds metrics.Gauge
+
+	NumTxs metrics.Gauge
+
+	//Time between current block and the previous block in seconds
+	BlockInterval metrics.Histogram
+}
```

Comment:
For some reason, you define a `Metrics` struct here with the methods `GetPrometheusMetrics` and `NilMetrics`, but you do the exact same thing in the `txpool` package.

Please define an interface and force these **specific implementations** to implement it.

Reply:
@zivkovicmilos, `GetPrometheusMetrics` & `NilMetrics` are more like constructor for individual metric subsystems, currently metrics type doesn't have any concrete methods, thats the reason why i didn't go with an interface.